Marlin/src/gcode/feature/pause/M701_M702.cpp

```diff
@@ -113,8 +117,12 @@ void GcodeSuite::M701() {
   #endif
 
   // Restore Z axis
-  if (park_point.z > 0)
-    do_blocking_move_to_z(_MAX(current_position.z - park_point.z, 0), feedRate_t(NOZZLE_PARK_Z_FEEDRATE));
+  if (park_point.z > 0) {
+    destination = current_position;
+    destination.z -= park_point.z;
```

Comment:
The lift could be smaller due to `NOMORE(destination.z, Z_MAX_POS);`. Z position has to be saved and restored.